CraftsmenLtd/BloodConnect
# I#160#165 Send Push Notification

Fixes #160 #165

## What kind of change does this PR introduce?

## How would you describe the changes in a changelog?
Added push notification system with user device token registration and notification delivery functionality through AWS SQS-SNS infrastructure.

## Related Issues/PRs
- Issue #160: Device token registration implementation
- Issue #165: Push notification delivery system implementation

## How to test

Testing steps:
1. Register device token:
   - Send POST request to `/notification/register` with device token
   - Verify token is saved in DynamoDB user profile

2. Send push notification:
   - Send POST request to `/notification/push/{userId}` with notification payload
   - Verify notification is queued in SQS
   - Confirm delivery through SNS topic

## Checklist
- [ ] I have added necessary documentation (OpenAPI specs)
- [x] I have followed the project's coding style (TypeScript, Clean Architecture)
- [x] Infrastructure is defined using Terraform following project standards
